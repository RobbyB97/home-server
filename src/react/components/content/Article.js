/*
   Standard Article
*/

import React from 'react'
import {Link, NavLink} from 'react-router-dom'
import {connect} from 'react-redux'

export const Article = (props) => (
    <div className="article">
        {props.title && 
            <h2 className="article__title">
                {props.title}
            </h2>
        }

        {props.text &&
            <p className="article__text">
                {props.text}
            </p>
        }

    </div>
)

export default connect()(Article)