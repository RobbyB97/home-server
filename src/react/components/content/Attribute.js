import React from 'react'
import {Link, NavLink} from 'react-router-dom'
import {connect} from 'react-redux'


const Attribute = (props) => (
    <div className="attribute">
        {
            props.image &&
                <div className="attribute__image">
                    <img src={props.image}/>
                </div>
        }
        {
            props.title && 
                <h2 className="attribute__title">
                    {props.title}
                </h2>
        }

        {   props.text &&
                <p className="attribute__text">
                    {props.text}
                </p>
        }

    </div>
)

export default Attribute