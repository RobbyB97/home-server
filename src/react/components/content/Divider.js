import React from 'react'

export class Divider extends React.Component {

    render() {
        return (
            <div className="divider">

            </div>
        )
    }
}


export default Divider