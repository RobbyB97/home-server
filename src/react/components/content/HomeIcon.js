import React from 'react'


export default class HomeIcon extends React.Component {

    render() {
        return (
            <div className="homeIcon">
                <a className="homeIcon__link" title={this.props.name} href={this.props.link}>
                    <img className="homeIcon__icon" src={this.props.icon} />
                </a>
            </div>
        )
    }
}