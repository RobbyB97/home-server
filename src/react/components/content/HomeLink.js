import React from 'react'
import {NavLink} from 'react-router-dom'


export default class HomeLink extends React.Component {

    render() {
        return (
            <div className='home-link'>
                {this.props.external ?
                    <a className="home-link__link" href={this.props.to}>
                        {this.props.icon &&
                            <img className="home-link__icon" src={this.props.icon} />
                        }

                        <p className="home-link__link-text">
                            {this.props.linkText ?
                                this.props.linkText
                            :
                                <p>HomeLink External</p>
                            }
                        </p>
                    </a>
                :
                    <NavLink className="home-link__link" to={this.props.to}>
                        {this.props.icon &&
                            <img className="home-link__icon" src={this.props.icon} />
                        }

                        <p className='home-link__link-text'>
                            {this.props.linkText ?
                                this.props.linkText
                            :
                                <p>HomeLink Internal</p>
                            }
                        </p>
                    </NavLink>
                }
            </div>
        )
    }
}