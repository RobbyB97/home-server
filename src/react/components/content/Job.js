import React from 'react'


export class Job extends React.Component {

    render() {
        return (
            <div className="job">
                <p className="job__title">
                    {this.props.title}
                </p>

                {this.props.text &&
                    <p className="job__text">
                        {this.props.text}
                    </p>
                }

                {this.props.image &&
                    <div className="job__image">
                        <img src={this.props.image} />
                    </div>
                }
            </div>
        )
    }
}


export default Job