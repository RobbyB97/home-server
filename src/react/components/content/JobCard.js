import React from 'react'

export default (props) => (
    <div className="jobcard">
        {props.title && 
            <h2 className="jobcard__title">
                {props.title}
            </h2>
        }

        {props.text &&
            <p className="jobcard__text">
                {props.text}
            </p>
        }

        {props.image &&
            <div className="jobcard__image">
                <img src={props.image} />
            </div>
        }

    </div>
)
