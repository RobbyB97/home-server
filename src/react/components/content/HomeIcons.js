import React from 'react'

import HomeIcon from './HomeIcon'


export default class HomeIcons extends React.Component {

    render() {
        return (
            <div className="homeIcons">
                {this.props.icons &&
                    this.props.icons.map((icon) => {
                        <HomeIcon {...icon} />
                    })
                }
            </div>
        )
    }
}