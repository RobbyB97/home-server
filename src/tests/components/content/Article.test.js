import React from 'react'
import {shallow} from 'enzyme'

import {Article} from '../../../react/components/content/Article'


let wrapper

beforeEach(() => {
    wrapper = shallow(<Article />)
})


test('Render Article with no props', () => {
    expect(wrapper).toMatchSnapshot()
})


test('Render Article with props', () => {
    wrapper = shallow(
        <Article 
            title='TITLE!'
            text='TEXT!'
        />
    )

    expect(wrapper).toMatchSnapshot()
})
