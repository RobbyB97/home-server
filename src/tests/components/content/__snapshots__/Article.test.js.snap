// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Article with no props 1`] = `
<div
  className="article"
/>
`;

exports[`Render Article with props 1`] = `
<div
  className="article"
>
  <h2
    className="article__title"
  >
    TITLE!
  </h2>
  <p
    className="article__text"
  >
    TEXT!
  </p>
</div>
`;
